weave/io_service.py

```diff
@@ -399,11 +399,25 @@ def unregister_client(
     async def handle_ensure_manifest(
         self, artifact_uri: str
     ) -> typing.Optional[artifact_wandb.WandbArtifactManifest]:
-        uri = artifact_wandb.WeaveWBArtifactURI.parse(artifact_uri)
+        print("inside io_service, ensure manifest()")
+        try:
```

Comment:
i think an alternative pattern here: `WeaveURI.parse` is what you are looking for, then if you want to be type safe, do an assertion on instance type. For example:
```
uri = WeaveURI.parse(artifact_uri)
if not isinstance(uri, (artifact_wandb.WeaveWBArtifactURI, artifact_wandb.WeaveWBArtifactByIDURI)):
   raise ... 
```